src/Testing/Commands/MakeExpectationCommand.php

```diff
@@ -479,36 +519,90 @@ protected function canReturnExpectation(ReflectionNamedType $returnType): bool
     /**
      * @return class-string|null
      */
-    private function getInputClass(string $basePath, Filesystem $filesystem): ?string
+    private function fileToClass(SplFileInfo $fileInfo): ?string
     {
-        /** @phpstan-var class-string|string $class */
-        $class = (string) $this->input->getArgument('class');
+        try {
+            $file = PhpFile::fromCode($fileInfo->getContents());
+        } catch (InvalidStateException|InvalidArgumentException $e) {
+            $this->writeError(sprintf('Error parse file [%s]: %s', $fileInfo->getPathname(), $e->getMessage()));
+            return null;
+        }
 
+        /** @phpstan-var array<class-string, ClassLike> $classes */
+        $classes = $file->getClasses();
+        if ($classes === []) {
+            $this->writeError(sprintf('Provided file does not contain any class [%s]', $fileInfo->getPathname()));
+            return null;
+        }
+
+        return $this->checkInterface(array_keys($classes)[0]);
+    }
+
+    /**
+     * @return class-string|null
+     */
+    private function checkInterface(string $class): ?string
+    {
+        if (class_exists($class) === false && interface_exists($class) === false) {
+            $this->writeError(sprintf('Provided class does not exists [%s]', $class));
+            return null;
+        }
+
+        return $class;
+    }
+
+    /**
+     * @return class-string|null
+     */
+    private function normalizeToClass(string $class, string $basePath, Filesystem $filesystem): ?string
+    {
         if (str_ends_with($class, '.php')) {
             $fullPath = $basePath . '/' . $class;
 
             if ($filesystem->exists($fullPath) === false) {
                 $this->writeError(sprintf('File does not exists at [%s]', $class));
                 return null;
             }
+            return $this->fileToClass(new SplFileInfo($fullPath, 'not used', 'not used'));
+        }
 
-            $file = PhpFile::fromCode($filesystem->get($fullPath));
+        return $this->checkInterface($class);
+    }
 
-            /** @phpstan-var array<class-string, ClassLike> $classes */
-            $classes = $file->getClasses();
-            if ($classes === []) {
-                $this->writeError(sprintf('Provided file does not contain any class [%s]', $class));
-                return null;
+    /**
+     * @return array<class-string>
+     */
+    private function findAllClasses(Finder $finder, CacheMeServiceContract $cacheMeService): array
+    {
+        $cacheKey = 'larastrict.make.expectation.interfaces';
+        $cachedInterfaces = $cacheMeService->get($cacheKey, static fn (): ?array => null);
```

Comment:
Použil jsem keš, pro další průchod, se neparsují již existující soubory.

Reply:
Tu keš asi nechápu :) Ty chceš to cachovat mezi volání commandu? Nevím jestli je to správně pokud tam nebude stejně check jestli implementace se nezměnila, a to mě připadá trochu out of scope (docela dost pracné).

Jinak za mě špatně použitá ta funkce - přesunout "ten" search do fn () a bud se spustí pokud cache není a nebo ne.